### ISSUE TITLE
Better exception message in case of a missing RealmMigration

### ISSUE THREAD

**Issue by cmelchior** [realm/realm-java#2542]
From this SO question: http://stackoverflow.com/questions/36330465/how-to-debug-realmmigrationneededexception-on-android/36358632#36358632

In case of migration errors with no configured `RealmMigration`, a rather unhelpful message is thrown:
```
Caused by: io.realm.exceptions.RealmMigrationNeededException: RealmMigration must be provided
```

It is of course true, but we should at the same time keep the underlying error, so people can see _why_ a migration is suddenly required.

**Comment by peerless2012** [comment]
How to solve this execption？

**Comment by beeender** [comment]
Fixed by #3482
